README.md

```diff
@@ -752,38 +769,67 @@ Apart from these new fields, the session history entries which correspond to `Ap
 
-An important consequence of this is that when iframes are involved, the back button may navigate through the joint session history, without changing the current _app history_ entry. For example, consider the following sequence:
+Example: consider the following setup.
 
-1. `https://example.com/start` loads
+1. `https://example.com/start` loads.
 1. The user navigates to `https://example.com/outer` by clicking a link. This page contains an iframe with `https://example.com/inner-start`.
+1. Code on `https://example.com/outer` calls `appHistory.pushNewEntry({ url: "/outer-pushed" })`
 1. The iframe navigates to `https://example.com/inner-end`.
 
-The app history list for the outer frame contains two entries:
-
-```
-1. https://example.com/start
-2. https://example.com/outer
-```
-
-The joint session session history contains three entries:
+The joint session session history contains four entries:
 
 ```
 A. https://example.com/start
 B. https://example.com/outer
    ┗ https://example.com/inner-start
-C. https://example.com/outer
+C. https://example.com/outer-pushed
+   ┗ https://example.com/inner-start
+D. https://example.com/outer-pushed
    ┗ https://example.com/inner-end
 ```
 
-The user's back button, as well as the `history.back()` API, will navigate the joint session history back to (B). However, they will have no effect on the app history list; that will stay on (2). Pressing the back button or calling `history.back()` a second time would then move the joint session history back to (A), and the app history list back to (1).
+The app history list (which also matches the existing spec's session history) for the outer frame looks like:
+
+```
+O1. https://example.com/start        (associated to A)
+O2. https://example.com/outer        (associated to B)
+O3. https://example.com/outer-pushed (associated to C and D)
+```
+
+The app history list for the inner frame looks like:
+
+```
+I1. https://example.com/inner-start  (associated to C)
+I2. https://example.com/inner-end    (associated to D)
+```
+
+Traversal operates on the joint session history, which means that it's possible to impact other frames. Continuing with our previous setup, and assuming the current entry in the joint session history is D, then
+
+- If code in the outer frame calls `appHistory.back()`, this will take us back to O2, and thus take the joint session history back to B. This means the inner frame will be navigated from `/inner-end` to `/inner-start`, changing its current app history entry from I2 to I1.
+
+- If code in the inner frame calls `appHistory.back()`, this will take us back to I1, and take the joint session history back to C. (This does not impact the outer frame.)
+
+- If code in either the inner frame or the outer frame calls `history.back()`, this will take the joint session history back to C, and thus update the inner frame's current app history entry from I2 to I1. (There is no impact on the outer frame.)
```

Comment:
This is an interesting outcome, because `C` isn't an entry in the iframe, it belongs to the parent.

I'm not saying it's wrong, but it's something that might need further documentation.

You could say: To traverse to a specific `appHistory` entry, joint history should traverse the least number of steps to make that history entry active.

That means you'd end up on a different item of joint session history depending on the direction of travel. I can't decide if that's intuitive or not.

The alternative is to _always_ travel to the step in joint history where that entry was created. That means it's consistent going back & forward, but may result in changing other frames more than necessary. In this example it would navigate the parent frame (and of course hit that fun bug).

Reply:
Yeah, I was going for least number of steps... hmm. I'll try to capture this and maybe open a tracking issue.

Reply:
Although I assume your comment is meant to apply to the second bullet, not the third. The third uses `history.back()`, not `appHistory.back()`.

Reply:
Ah yes sorry